apps/meteor/app/lib/server/lib/notifyListenerOnRoomChanges.ts

```diff
@@ -0,0 +1,27 @@
+import { api, dbWatchersDisabled } from '@rocket.chat/core-services';
+import type { IRoom } from '@rocket.chat/core-typings';
+import { Rooms } from '@rocket.chat/models';
+
+type ClientAction = 'inserted' | 'updated' | 'removed';
+
+export async function notifyListenerOnRoomChanges(
+	rid: IRoom['_id'],
+	clientAction: ClientAction = 'updated',
+	existingRoomData?: IRoom,
+	args?: { [key: string]: any },
+): Promise<void> {
+	if (!dbWatchersDisabled) return;
+
+	const room = existingRoomData || (await Rooms.findOneById(rid));
+
+	if (room) {
+		void api.broadcast('watch.rooms', { clientAction, room: { ...room, ...args } });
+	}
```

Comment:
just a picky code style comment, the main operation of the function should not be inside a condition:
```suggestion
	if (!room) {
		return;
	}
	void api.broadcast('watch.rooms', { clientAction, room: { ...room, ...args } });
```